src/machine/machine-varlink.c

```diff
@@ -570,3 +571,191 @@ int vl_method_open(sd_varlink *link, sd_json_variant *parameters, sd_varlink_met
 
         return sd_varlink_reply(link, v);
 }
+
+typedef struct MachineCopyParameters {
+        const char *name;
+        PidRef pidref;
+        char *src, *dest;
+        bool replace;
+} MachineCopyParameters;
+
+static void machine_copy_paramaters_done(MachineCopyParameters *p) {
+        assert(p);
+
+        pidref_done(&p->pidref);
+        free(p->src);
+        free(p->dest);
+}
+
+static int copy_done(Operation *operation, int ret, sd_bus_error *error) {
+        assert(operation);
+        assert(operation->link);
+
+        // TODO(ikruglov): maybe just leaving a plain errno in response?
```

Comment:
So I would only do a proper custom error object if we can include more useful information such as the filename. Otherwise I would just go for `sd_varlink_error_errno()` for now.

Reply:
But maybe something to bring up at the biweekly as well on how we can propagate more useful errors.

Reply:
the idea here was to save people from looking up what errno: X means for common errors such as NoSuchFile, FileExists, PermissionDenied, etc

Reply:
Yeah we should add the human readable form of the error to the error struct. But we can add this later I think after we discuss it in the biweekly? Should just be another field in the error object so its a backwards compatible change.